components/Events/Filters.js

```diff
@@ -0,0 +1,236 @@
+import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState, useReducer, React } from 'react';
+import events from '../../data/event';
+import styles from '../../styles/components/Events/Filters.module.scss';
+
+export default function filters(props) {
+  function reducer(state, action) {
+    switch (action.type) {
+      case 'comms':
+        return { ...state, comms: { ...state.comms, [action.payload.key]: action.payload.val } };
+      case 'days':
+        return { ...state, days: { ...state.days, [action.payload.key]: action.payload.val } };
+      case 'times':
+        return { ...state, times: { ...state.times, [action.payload.key]: action.payload.val } };
+      case 'loc':
+        return { ...state, loc: { ...state.loc, [action.payload.key]: action.payload.val } };
+      default:
+        throw new Error();
+    }
+  }
+  const initialState = {
+    comms: {
+      'studio': false,
+      'icpc': false,
+      'design': false,
+      'cyber': false,
+      'teach-la': false,
+      'w': false,
+      'ai': false,
+      'hack': false,
+      'board': false,
+    },
+    days: {
+      Sunday: false,
+      Monday: false,
+      Tuesday: false,
+      Wednesday: false,
+      Thursday: false,
+      Friday: false,
+      Saturday: false,
+    },
+    times: {
+      from: '',
+      to: '',
+    },
+    loc: {
+      'Online': false,
+      'In-Person': false,
+    },
+  };
+  const [filterVals, dispatch] = useReducer(reducer, initialState);
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const dateToMinutes = (date) => {
+    const itemMins =
+      date.getHours() * 60 + date.getMinutes();
+    return itemMins;
+  };
+  const filterTime = (filterEvents) => {
+    const isFilterTimeEmpty = Object.keys(filterVals.times).every(
+      (time) => filterVals.times[time] === '',
+    );
+    if (!isFilterTimeEmpty) {
+      const filterFromMins =
+        filterVals.times.from === '' ? 0 : // 12:00 AM if empty
+          parseInt(filterVals.times.from.split(':')[0]) * 60 +
+          parseInt(filterVals.times.from.split(':')[1]);
+      const filterToMins =
+        filterVals.times.to === '' ? 1440 :  // 11:59 PM if empty
+          parseInt(filterVals.times.to.split(':')[0]) * 60 +
+          parseInt(filterVals.times.to.split(':')[1]);
+
+      return filterEvents.filter((item) => {
+        const itemStartMins = dateToMinutes(new Date(item.start));
+        const itemEndMins = dateToMinutes(new Date(item.end));
+        return itemStartMins >= filterFromMins && itemEndMins <= filterToMins;
+      });
+    }
+    return filterEvents;
+  };
+  const filterComm = (filterEvents) => {
```

Comment:
nit: use new lines to cleanly separate and make code easier to read
```suggestion

  const filterComm = (filterEvents) => {
```